apps/meteor/app/lib/server/functions/addUserToRoom.ts

```diff
@@ -112,9 +112,11 @@ export const addUserToRoom = function (
 		});
 	}
 
-	if (room.teamMain && room.teamId && inviter) {
+	if (room.teamMain && room.teamId) {
+		const inviterId = inviter ? inviter._id : room.u._id;
+
 		// if user is joining to main team channel, create a membership
-		Promise.await(Team.addMember(inviter, userToBeAdded._id, room.teamId));
+		Promise.await(Team.addMember(inviterId, userToBeAdded._id, room.teamId));
```

Comment:
The first param of the `addMember` method should be an object containing the `_id` field. So here we should use `inviter || userToBeAdded` as the first param.
Using the `room.u._id` would mean that the room owner is the inviter, which is not always the case.

Reply:
Updated!